updates mutate and validate web hook configs

zanuka opened the pull request:
**What this PR does / why we need it**:

This PR adds `MutatingWebhookConfiguration` and `ValidatingWebhookConfiguration` to the Helm installation and uninstallation order. These resources are placed at the end of the installation order (after APIService) and at the beginning of the uninstallation order, which matches the expected behavior that webhook configurations should be installed after workloads like deployments and pods.

This addresses issue #30617 which requested clarification on the installation order of these Kubernetes resources. While the original issue requested documentation updates, these webhooks weren't explicitly included in the code. 

Scott note:
fixes #5642

By adding them we get the following:
- webhooks are installed after workloads, preventing potential interference
- provide predictable installation/uninstallation behavior
- enable accurate documentation of an intentional ordering rather than implicit behavior

**Special notes**:

This change only affects the ordering of resources during installation and uninstallation. The existing tests have been updated to account for these additions, and all tests pass.

**If applicable**:

[x] this PR contains user facing changes (the docs needed label should be applied if so)
[x] this PR contains unit tests
[ ] this PR has been tested for backwards compatibility ( not sure )

Note: After this PR is merged, a follow-up PR to the helm-www repository will be needed to update the installation guide documentation to reflect these changes.

zanuka committed:
updates mutate and validate web hook configs

Signed-off-by: Mike Delucchi <git@zanuka.com>

zanuka commented:
supplemental PR for helm-www docs here - https://github.com/helm/helm-www/pull/1710

robertsirc reviewed (dismissed):
LGTM

jvoravong commented on pkg/release/util/kind_sorter.go:
```diff
@@ -65,12 +65,16 @@ var InstallOrder KindSortOrder = []string{
 	"IngressClass",
 	"Ingress",
 	"APIService",
+	"MutatingWebhookConfiguration",
```
For my use cases, I’d would request to adjust the order of operations slightly.

- **Lines 67-69:**  
  ```yaml
  ValidatingWebhookConfiguration
  MutatingWebhookConfiguration
  APIService
  ```  
- **Lines 76-78:**  
  ```yaml
  APIService
  MutatingWebhookConfiguration
  ValidatingWebhookConfiguration
  ```  

For example, when an incoming request interacts with my webhooks, the workflow looks kind of like this sequence:  

```
Request -> APIService -> ValidatingWebhook -> MutatingWebhook -> API Server
```
For example, when creating a new pod:  
1. The request reaches **APIService v1alpha1.opentelemetry.io**.  
2. The **ValidatingWebhook** determines whether the pod should be mutated.  
3. If allowed, the **MutatingWebhook** mutates the request for observability instrumentation purposes.
4. The pod request then proceeds through normal Kubernetes processing for creating a pod.

zanuka replied:
@jvoravong makes sense to me, will adjust

zanuka replied:
updated

TerryHowe replied:
In Kubernetes, the validation webhook determines if the request is valid *after* the mutating webhooks.

zanuka committed:
changes order of operations

Signed-off-by: Mike Delucchi <git@zanuka.com>

zanuka committed:
fixes kind_sorter_test to match new order

Signed-off-by: Mike Delucchi <git@zanuka.com>

zanuka commented:
@TerryHowe DCO signoff issue fixed

zanuka committed:
fix: correct webhook order to match Kubernetes admission flow

Place APIService before webhooks, with MutatingWebhookConfiguration
before ValidatingWebhookConfiguration to match standard admission control.

Signed-off-by: Mike Delucchi <git@zanuka.com>

zanuka commented:
@TerryHowe just updated the order

scottrigby reviewed (approved):
Nice one. LGTM
